Mathlib/Combinatorics/SimpleGraph/Diam.lean

```diff
@@ -228,4 +276,77 @@ lemma connected_iff_diam_ne_zero [Fintype α] [Nontrivial α] : G.Connected ↔
 
 end diam
 
+section radius
+
+/-- The radius of a simple graph is the minimum eccentricity of any vertex. -/
+noncomputable def radius (G : SimpleGraph α) : ℕ∞ :=
+  ⨅ u, G.eccent u
+
+/-- The center of a simple graph is the set of vertices with eccentricity equal to the radius. -/
+def center (G : SimpleGraph α) : Set α :=
+  {u | G.eccent u = G.radius}
+
+lemma center_nonempty [Nonempty α] : G.center.Nonempty :=
+  ENat.exists_eq_iInf G.eccent
+
+lemma radius_eq_iInf_iSup_edist : G.radius = ⨅ u, ⨆ v, G.edist u v :=
+  rfl
+
+lemma radius_le_eccent (u : α) : G.radius ≤ G.eccent u :=
+  iInf_le G.eccent u
+
+lemma radius_eq_top_of_not_connected (h : ¬G.Connected) :
+    G.radius = ⊤ := by
+  simp [radius, eccent_eq_top_of_not_connected h]
+
+lemma radius_eq_top_of_isEmpty [IsEmpty α] : G.radius = ⊤ :=
+  iInf_of_empty G.eccent
+
+lemma radius_ne_zero_of_nontrivial [Nontrivial α] : G.radius ≠ 0 := by
+  rw [← ENat.one_le_iff_ne_zero]
+  apply le_iInf
+  simp [ENat.one_le_iff_ne_zero, G.eccent_ne_zero]
+
+lemma radius_eq_zero_iff : G.radius = 0 ↔ Nonempty α ∧ Subsingleton α := by
+  refine ⟨fun h ↦ ⟨?_, ?_⟩, fun ⟨_, _⟩ ↦ ?_⟩
+  · contrapose! h
+    simp [radius, not_nonempty_iff.mp h]
+  · contrapose! h
+    simp [not_subsingleton_iff_nontrivial.mp h, radius_ne_zero_of_nontrivial]
+  · rw [radius, ENat.iInf_eq_zero]
+    use Classical.ofNonempty
+    simpa [eccent] using Subsingleton.elim _
+
+lemma exists_edist_eq_radius_of_finite [Nonempty α] [Finite α] :
+    ∃ u v, G.edist u v = G.radius := by
+  obtain ⟨w, hw⟩ := G.center_nonempty
+  obtain ⟨v, hv⟩ := G.exists_edist_eq_eccent_of_finite w
+  use w, v
+  rw [hv, hw]
+
+lemma radius_le_ediam [Nonempty α] : G.radius ≤ G.ediam :=
+  iInf_le_iSup
+
+lemma ediam_le_radius_mul_two [Finite α] : G.ediam ≤ 2 * G.radius := by
+  cases isEmpty_or_nonempty α
+  · rw [radius_eq_top_of_isEmpty]
+    exact le_top
+  · by_cases h : G.Connected
+    · obtain ⟨u, v, h⟩ := G.exists_edist_eq_ediam_of_ne_top (connected_iff_ediam_ne_top.mp h)
+      obtain ⟨w, hw⟩ := G.center_nonempty
+      exact le_trans (h.symm ▸ G.edist_triangle (v := w))
+      <| two_mul G.radius ▸ hw ▸ add_le_add (G.edist_comm ▸ G.edist_le_ecc w u) (G.edist_le_ecc w v)
```

Comment:
This is pretty unpleasant, let's put this back to a tactic mode proof; maybe starting with `refine le_trans (h.symm ▸ G.edist_triangle (v := w)) ?_`. (Maybe you don't need the .symm too?)